### PR TITLE
feat: Add ResourceProvider for environment variables with scrubbing

### LINKED ISSUE #21: Collect environment variables and make them redactable via LUMIGO_SECRET_MASKING_REGEX env var
Add a resource to the Lumigo distro that collects as resource attributes the key-value pairs of environment variables (as in `System.getenv()`). The structure of the resource attribute must follow the implementation of the [Lumigo OpenTelemetry Python distro](https://github.com/lumigo-io/opentelemetry-python-distro/blob/943847ffac919f7fe3a60adce90b3640a151786e/src/lumigo_opentelemetry/resources/detectors.py#L64).

The Lumigo OpenTelemetry Java distro must also implement the `LUMIGO_SECRET_MASKING_REGEX` and `LUMIGO_SECRET_MASKING_REGEX_ENVIRONMENT` environment variables to follow the implementation found in all other major Lumigo tracers (see, e.g., https://github.com/lumigo-io/lumigo-node#configuration).

### PR DESCRIPTION
**Pull request by kenfinnigan**
Fixes #21

### CODE REVIEW

File: custom/src/main/java/io/lumigo/javaagent/common/SecretScrubber.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    final List<Pattern> patterns;
+    if (!Strings.isBlank(regExps)) {
+      patterns = new ArrayList<>();
+      try (JsonParser parser = JSON_FACTORY.createParser(regExps)) {
+        if (!parser.nextToken().isStructStart()
+            || !JsonToken.START_ARRAY.equals(parser.currentToken())) {
+          throw new IllegalArgumentException();
+        }
+        while (!JsonToken.END_ARRAY.equals(parser.nextToken())) {
+          if (parser.currentToken().equals(JsonToken.VALUE_STRING)) {
```
Comment (mmanciop) [inline custom/src/main/java/io/lumigo/javaagent/common/SecretScrubber.java:81]: If it's not a string, we should fall back to the default secret scrubbing patterns. If I read the code right, now we would silently skip the horror.

File: custom/src/test/java/io/lumigo/javaagent/common/SecretScrubberTest.java [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import java.time.Duration;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class SecretScrubberTest {
+  @Test
+  public void testEmptyEnvironmentMasking() {
+    ConfigProperties config =
```
Comment (mmanciop) [inline custom/src/test/java/io/lumigo/javaagent/common/SecretScrubberTest.java:32]: Maybe mockito can make this less verbose?

**Review by mmanciop — DISMISSED** [review]
I did not do a proper pass with PR checkout and all, but I think there's enough for a quick iteration before I do

File: custom/src/main/java/io/lumigo/javaagent/common/ProcessEnvironmentScrubber.java [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+
+public class ProcessEnvironmentScrubber extends SecretScrubber {
+  public static final Logger LOGGER = Logger.getLogger(ProcessEnvironmentScrubber.class.getName());
+
+  public static final String LUMIGO_SECRET_MASKING_REGEX_ENVIRONMENT =
+      "lumigo.secret.masking.regex.environment";
+
+  final String regExps;
```
[the remark this answers is not in the document]
Comment (kenfinnigan) [reply custom/src/main/java/io/lumigo/javaagent/common/ProcessEnvironmentScrubber.java:35]: It's used in `SecretScrubberTest` via package. We can alternatively make the field private and have a package getter

File: custom/src/main/java/io/lumigo/javaagent/common/ParseExpressionResult.java [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+import java.util.List;
+import java.util.regex.Pattern;
+
+public class ParseExpressionResult {
+  private String sourceEnvVar;
+  private String regularExpressions;
+  private List<Pattern> expressionPatterns;
+
+  public ParseExpressionResult(List<Pattern> expressionPatterns) {
```
[the remark this answers is not in the document]
Comment (kenfinnigan) [reply custom/src/main/java/io/lumigo/javaagent/common/ParseExpressionResult.java:28]: Sure, thanks

File: custom/src/main/java/io/lumigo/javaagent/common/ProcessEnvironmentScrubber.java [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+          generator.writeStringField(entry.getKey(), entry.getValue());
+        }
+      }
+
+      generator.writeEndObject();
+    } catch (IOException e) {
+      throw new RuntimeException(e);
+    }
+
+    return writer.toString().replaceAll("\"", "\\\\\"");
```
[the remark this answers is not in the document]
Comment (kenfinnigan) [reply custom/src/main/java/io/lumigo/javaagent/common/ProcessEnvironmentScrubber.java:69]: I'll add a comment. It's essentially taking the JSON string and escaping it as what's returned from the writer is not properly escaped.

In trying to get the writer to properly escape the string, there appeared to only be a means of doing that if writing the JSON to a file

File: custom/src/test/java/io/lumigo/javaagent/common/SecretScrubberTest.java (hunk 2 above)
Comment (kenfinnigan) [reply custom/src/test/java/io/lumigo/javaagent/common/SecretScrubberTest.java]: Now added

**Review by mmanciop — APPROVED** [review]

**Pull request merged**